packages/opentelemetry-tracing/src/export/BatchSpanProcessor.ts

```diff
@@ -92,49 +102,84 @@ export class BatchSpanProcessor implements SpanProcessor {
 
   /** Add a span in the buffer. */
   private _addToBuffer(span: ReadableSpan) {
+    if (this._finishedSpans.length >= this._maxQueueSize) {
+      // limit reached, drop span
+      return;
+    }
     this._finishedSpans.push(span);
     this._maybeStartTimer();
-    if (this._finishedSpans.length > this._bufferSize) {
-      this._flush().catch(e => {
-        globalErrorHandler(e);
-      });
-    }
   }
 
-  /** Send the span data list to exporter */
+  /**
+   * Send all spans to the exporter respecting the batch size limit
```

Comment:
Why have 2 flushes? The difference between this and _flush is not really explained.

Reply:
Previously `BatchSpanProcessor` if you ended 1mln spans it will try to export 1mln spans in one export. But spec says the batch has limit so now the flush is exporting preserving the limit. When you call shutdown we have to export all spans. So `flashAll` will simply call flash as many times as needed and when all of this will finish it will be finally resolved. For example you have 800 spans in queue it will call `flash` 2 times, first will take 512 spans and the 2nd will take remaining 288 spans. Both of them will be exported at the same time creating 2 requests. Once both of the requests are finished `flashAll` will be resolved.
We also have another limit which is the maximum queue size so if you were to export 1mln spans it will just allow first 2048 spans to be added to the queue - the rest will be dropped. And then each cycle of batch export will be able to export 512 spans with every interval. So to be able to export all 2048 span you will need 4 export cycles.

Reply:
Can we rename these functions to `flushOneBatch` and `flushAll` to make it more clear?